### PR TITLE
Cisco IOS (cisco.router and cisco.switch) new rules

### PR DESCRIPTION
**Pull request by nnovaes**
## Description

new parsing rules for cisco.router and cisco.switch. The old version of this processor needs some rework. However, there are functioning bits of it that i have preserved, since they kind of work. the new rules provide some good foundation for future "full" parsing and also covers bgp and interface up/down msgs. the lookup database for translate filters is static.

### CODE REVIEW

**Comment by nnovaes** [comment]
@brian-grabau @verbecee 
 I have moved the interface fields under observer, however i don't necessarily agree these fields should be there. i've updated https://github.com/elastic/ecs/issues/610 with this issue, others have been discussing as well. perhaps if/when it gets resolved we should revisit this field mapping.

**Review by KrishnanandSingh — CHANGES_REQUESTED** [review]
Indentations are not uniform. Please use 2 spaces to indent everywhere.

File: config/processors/syslog_log_audit_cisco.router.conf [hunk 1]
```diff
@@ -36,9 +38,255 @@ filter {
...
+  # 2. General cisco log format is 
+  # {timesdtamp} {facility} {severity} {mnemonic} {description}
+  # seq no:timestamp: %facility-severity-MNEMONIC:description
+
+  grok {
+    id => "cisco.router-mutate-grok-message"
+    tag_on_failure => "_grokparsefailure_cisco.router-mutate-grok-message"
+    match => { 
+        "message" => [
+          "%{GREEDYDATA:[tmp][ingest_timestamp]}\s\{name=%{GREEDYDATA:[tmp][logsource]}\}\s<%{NUMBER:syslog_pri}>%{NUMBER:[event.id]}:\s%{GREEDYDATA:[tmp][device_timestamp]}:\s\%%{WORD:[tmp][facility]}-%{WORD:[tmp][severity]}-%{WORD:[tmp][mnemonic]}:\s%{GREEDYDATA:[tmp][msg]}",
```
[the remark this answers is not in the document]
Comment (nnovaes) [reply config/processors/syslog_log_audit_cisco.router.conf:77]: @KrishnanandSingh ,

If we are parsing all syslog pri headers but not removing them from the message for the downstream filters (which makes sense) , i believe it is still relevant to have those fields in groks/dissects throughout configs, however we'd need to change the field the grok captures it, so in this case i'll need to replace <%{NUMBER:syslog_pri}> with <%{NUMBER}>. i'll make that change

[the remark this answers is not in the document]
Comment (nnovaes) [reply config/processors/syslog_log_audit_cisco.router.conf]: done. removed in #4899c1d

**Comment by nnovaes** [comment]
@KrishnanandSingh  ready to merge

File: config/processors/syslog_log_audit_cisco.router.conf [hunk 2]
```diff
@@ -6,18 +6,20 @@ input {
   }
 }
 filter {
+
   mutate {
     add_field => { "observer.vendor" => "cisco" }
     add_field => { "event.module" => "cisco.router" }
     add_field => { "observer.product" => "cisco.router" }
     add_field => { "observer.type" => "router" }
     # log.source.hostname does not exist in ECS. this field comes from the relay. because it may break something we'll just copy it to the proper field name before someone addresses it
```
Comment (KrishnanandSingh) [inline config/processors/syslog_log_audit_cisco.router.conf]: this comment should be removed now as we are not copying it.

**Review by KrishnanandSingh — APPROVED** [review]

**Pull request merged**